tsdb/block.go

```diff
@@ -71,11 +73,9 @@ type IndexReader interface {
 	// LabelValues returns possible label values which may not be sorted.
 	LabelValues(ctx context.Context, name string, matchers ...*labels.Matcher) ([]string, error)
 
-	// Postings returns the postings list iterator for the label pairs.
-	// The Postings here contain the offsets to the series inside the index.
-	// Found IDs are not strictly required to point to a valid Series, e.g.
-	// during background garbage collections.
-	Postings(ctx context.Context, name string, values ...string) (index.Postings, error)
+	// PostingsForMatcher returns a sorted iterator over postings having a label matching the provided label matcher.
```

Comment:
Hm, it's bit odd to move Postings to PostingsReader but leaving PostingsForMatchers. The latter also feels like reading postings (just with extra filtering), so why not moving both ot index pkg for consistency (or leaving both here)

Reply:
Logically, I agree that `PostingsForMatchers` would make sense in `PostingsReader` too. I just moved `Postings` because that's the method that's needed in the new sub-interface. `PostingsReader`, and its method `Postings`, are required by `index.MemPostings.PostingsForMatcher` in case that wasn't clear.

Even if `Postings` isn't required in the `PostingsReader` interface, do you still think I should move that method there too?

Reply:
I double checked the code and I see indeed the point for existence of sub interface. Definitely let's not shovel the PostingsForMatcher there, if anything it could be separate interface. However no one would need that sub interface alone for now, so it's not worth it.

It's fine for now, I guess more clean would be to literally keep sub interface (just make it private) but explicitly define (again) method here? (this way the subinterface is scoped down only for one method that requires it). I am fine both ways.

Reply:
If we want to move `PostingsForMatchers` method to `PostingsReader`, then we need a `PostingsReader()` method on the `BlockReader` interface. This will break a lot of code, but might be a good long-term improvement because the `IndexReader` is getting quite beefy and is not really idiomatic by go standards.

Reply:
@fpetkovski is it technically necessary though to add a `PostingsReader` method to `BlockReader`, considering `IndexReader` is a superset of `PostingsReader`?

Are you suggesting to not have `IndexReader` include `PostingsReader`?

Reply:
I don't think it's necessary to make further changes for now. Ideally IndexReader and PostingsReader would be two separate interfaces and one would not be embedded into the other.

Reply:
Gotcha, thanks for clarifying your thinking.